### PR TITLE
Add associated files refiner

### PR DESCRIPTION
**Pull request by medariox**
- [x] PR is based on the DEVELOP branch
- [x] Don't send big changes all at once. Split up big PRs into multiple smaller PRs that are easier to manage and review
- [x] Read the [contribution guide](https://github.com/pymedusa/Medusa/blob/master/.github/CONTRIBUTING.md)

### CODE REVIEW

File: medusa/post_processor.py [hunk 1]
```diff
@@ -165,101 +146,102 @@ def _compare_file_size(self, existing_file):
             ))
             return self.EXISTS_LARGER if new_size < old_size else self.EXISTS_SMALLER
 
-    def list_associated_files(self, filepath, base_name_only=False, subtitles_only=False, subfolders=False):
+    def list_associated_files(self, filepath, subfolders=None, subtitles_only=None, refine=None):
```
Comment (fernandog) [inline medusa/post_processor.py]: @medariox why refine=None instead of False?

File: medusa/post_processor.py [hunk 2]
```diff
@@ -165,101 +146,102 @@ def _compare_file_size(self, existing_file):
...
-        :param subtitles_only: list only subtitles
-        :param subfolders: check subfolders while listing files
+        :param filepath: path of the file to check for associated files
+        :param subfolders: also check subfolders while searching files
+        :param subtitles_only: list only associated subtitles
+        :param refine: refine the associated files with additional options
         :return: A list containing all files which are associated to the given file
         """
-        files = self._search_files(filepath, subfolders=subfolders, base_name_only=base_name_only)
+        files = self._search_files(filepath, subfolders=subfolders, basename_only=True)
```
Comment (fernandog) [inline medusa/post_processor.py]: @medariox this breaks when a .nfo is associated to the .rar base name

As you are only searching files that are equal .mkv base name, it won't find the .nfo as a file

Comment (fernandog) [inline medusa/post_processor.py]: Test p6 is failing proving that

Comment (medariox) [inline medusa/post_processor.py]: As you can see the PR is still in progress. That means I'm still working on it.

File: medusa/post_processor.py [hunk 3]
```diff
@@ -302,7 +302,7 @@ def _delete(self, files, associated_files=False):
 
         # also delete associated files, works only for 1 file
         if associated_files and len(files) == 1:
-            file_list += self.list_associated_files(files, subfolders=True)
+            file_list += self.list_associated_files(files[0], subfolders=True)
```
Comment (fernandog) [inline medusa/post_processor.py]: Isn't this the reason of folder not being deleted in root folder as its only deletes one associated file ?

`# also delete associated files, works only for 1 file`

Comment (medariox) [inline medusa/post_processor.py]: Which issue would that be?

Comment (fernandog) [inline medusa/post_processor.py]: @wimpyrbx do you still have folders in PP not being deleted because of leftover files?

Comment (fernandog) [inline medusa/post_processor.py]: @medariox nevermind. we pass associated_files=true in `_delete`  to delete existing files in library when we are replacing.

**Comment by fernandog** [comment]
@medariox so far so good. PPed 8 files with subs (postpone)

**Comment by fernandog** [comment]
@medariox i didn't test the nfo deleting on rename. this PR also fix that issue right?

**Comment by medariox** [comment]
@fernandog Yes.

**Pull request merged**